letsencrypt-nginx/letsencrypt_nginx/configurator.py

```diff
@@ -598,6 +613,10 @@ def nginx_restart(nginx_ctl, nginx_conf="/etc/nginx.conf"):
     except (OSError, ValueError):
         logger.fatal("Nginx Restart Failed - Please Check the Configuration")
         sys.exit(1)
+    # Nginx can take a moment to recognize a newly added TLS SNI servername, so sleep
+    # for a second. TODO: Check for expected servername and loop until it
+    # appears or return an error if looping too long.
+    time.sleep(1)
```

Comment:
(a) maybe just connect to the domain in question?
(b) you can use `acme.crypto_util.probe_sni` for that :)

(a) and (b) together: `probe_sni(domain, socket.gethostbyname(domain), config.dvsni_port)`